lib/polymorphic_integer_type/belongs_to_polymorphic_association_extension.rb

```diff
@@ -1,19 +1,25 @@
-module ActiveRecord
-  module Associations
-    class BelongsToPolymorphicAssociation < BelongsToAssociation
-      private
+module PolymorphicIntegerType
+  module BelongsToPolymorphicAssociationExtension
+    private
 
-      if Gem::Version.new(ActiveRecord::VERSION::STRING) < Gem::Version.new("6.1")
-        def replace_keys(record)
-          super
+    if Gem::Version.new(ActiveRecord::VERSION::STRING) < Gem::Version.new("6.1")
+      def replace_keys(record)
+        super
+
+        if reflection.options[:integer_type] || reflection.options[:polymorphic].is_a?(Hash)
           owner[reflection.foreign_type] = record.class.base_class unless record.nil?
         end
-      elsif
-        def replace_keys(record, force: false)
-          super
+      end
+    else
+      def replace_keys(record, force: false)
+        super
+
+        if reflection.options[:integer_type] || reflection.options[:polymorphic].is_a?(Hash)
           owner[reflection.foreign_type] = record.class.base_class unless record.nil?
         end
       end
     end
   end
 end
+
+ActiveRecord::Associations::BelongsToPolymorphicAssociation.prepend(PolymorphicIntegerType::BelongsToPolymorphicAssociationExtension)
```

Comment:
In order to have `super` above appropriately use the Rails-defined method in `ActiveRecord::Associations::BelongsToPolymorphicAssociation`, we need to prepend this code instead of override the method altogether.